genie-web/src/integTest/java/com/netflix/genie/web/data/services/jpa/JpaJobPersistenceServiceImplIntegrationTest.java

```diff
@@ -734,6 +735,76 @@ public void canGetJobArchiveLocation() throws GenieNotFoundException {
         );
     }
 
+    /**
+     * Test {@link JpaJobPersistenceServiceImpl#getFinishedJob(String)}.
+     *
+     * @throws GenieNotFoundException if the test works
+     */
+    @Test(expected = GenieNotFoundException.class)
+    public void canGetFinishedJobNonExistent() throws GenieNotFoundException {
+        this.jobPersistenceService.getFinishedJob(UUID.randomUUID().toString());
+        Assert.fail();
+    }
+
+    /**
+     * Test {@link JpaJobPersistenceServiceImpl#getFinishedJob(String)}.
+     *
+     * @throws GenieNotFoundException if the test is broken
+     */
+    @Test(expected = GenieInvalidStatusException.class)
+    public void canGetFinishedJobNotFinished() throws GenieNotFoundException {
+        this.jobPersistenceService.getFinishedJob(JOB_3_ID);
+        Assert.fail();
+    }
+
+    /**
+     * Test {@link JpaJobPersistenceServiceImpl#getFinishedJob(String)}.
+     *
+     * @throws GenieNotFoundException if the test is broken
+     */
+    @Test
+    public void canGetFinishedJob() throws GenieNotFoundException {
+        final FinishedJob finishedJob = this.jobPersistenceService.getFinishedJob(JOB_1_ID);
+        Assert.assertNotNull(finishedJob);
+
+        Assert.assertThat(finishedJob.getUniqueId(), Matchers.is(JOB_1_ID));
```

Comment:
Just FYI I've started using the `assertj` libraries for stuff like this as I feel it's a bit cleaner e.g.

`Assertions.assertThat(finishedJob.getUniqueId).isEqualTo(JOB_1_ID)`... saves a bit of writing and has a more fluid API from what I've seen so far